src/App.js

```diff
@@ -19,23 +20,96 @@ const ROUTES = {
   MANAGEMENT: '/management',
 };
 
+const LANDING_IMG = "https://storage.googleapis.com/barnard-public-assets/Barnard%20lock%20up%202.1.png";
+
 class App extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.authUtils_ = new AuthUtils();
+
+    this.state = {
+      email: null,
+      authInitializing: true,
+    };
+  }
+
+  componentDidMount() {
+    this.authUtils_.getFirebaseAuth().onAuthStateChanged((user) => {
+      if (user) {
+        this.setState({email: user.email, authInitializing: false});
+      } else {
+        this.setState({email: null, authInitializing: false});
```

Comment:
Could be more succinct without the branch: setState({email: user ? user.email : null, ...)

Reply:
fixed.